packages/chain-bridge/src/adapters/StargateAdapter.sol

```diff
@@ -818,6 +1350,153 @@ contract StargateAdapter is Ownable, IBridgeAdapter, ILayerZeroComposer {
         );
     }
 
+    /**
+     * @dev Try to deposit assets to FleetCommander without reverting on failure
+     */
+    function _tryDepositToFleetCommander(
+        address fleetCommander,
+        address asset,
+        uint256 amount,
+        address shareRecipient,
+        bytes memory referralCode,
+        bytes32 operationId,
+        address, // originalUser
+        address, // sourceAsset
+        uint16 sourceChainId
+    ) internal returns (bool success) {
+        // Validate FleetCommander is active through HarborCommand
+        if (
+            !IHarborCommandMinimal(harborCommand).activeFleetCommanders(
+                fleetCommander
+            )
+        ) {
+            emit CrossChainFleetDepositFailed(
+                operationId,
+                fleetCommander,
+                asset,
+                amount,
+                "FleetCommander not active"
+            );
+            return false;
+        }
+
+        // Validate FleetCommander supports the asset
+        try IFleetCommanderMinimal(fleetCommander).asset() returns (
+            address fleetAsset
+        ) {
+            if (fleetAsset != asset) {
+                emit CrossChainFleetDepositFailed(
+                    operationId,
+                    fleetCommander,
+                    asset,
+                    amount,
+                    "Asset mismatch"
+                );
+                return false;
+            }
+        } catch {
+            emit CrossChainFleetDepositFailed(
+                operationId,
+                fleetCommander,
+                asset,
+                amount,
+                "Asset check failed"
+            );
+            return false;
+        }
+
+        // Check deposit limits
+        try
+            IFleetCommanderMinimal(fleetCommander).maxDeposit(address(this))
```

Comment:
i don't like the `if we can't check the limit, just ignore it and try anyway.` pattern

Reply:
Removed